### PR TITLE
hopefully fixes DPTNet this time 

### PR DESCRIPTION
**Pull request by popcornell**
see https://github.com/asteroid-team/asteroid/issues/380

### CODE REVIEW

File: asteroid/masknn/attention.py [hunk 1]
```diff
@@ -45,12 +45,6 @@ def __init__(
 
         self.mha = MultiheadAttention(embed_dim, n_heads, dropout=dropout)
         self.dropout = nn.Dropout(dropout)
-        self.pos_wise_ff = nn.Sequential(
```
Comment (mpariente) [inline asteroid/masknn/attention.py]: Why did you decide to remove it?

Comment (popcornell) [reply asteroid/masknn/attention.py]: Because I got confused it is not in DPTNet sorry I confused also you. 
Also we can implement the whole with two permutations. This is not efficient

**Review by SungFeng-Huang — APPROVED** [review]
Seems to be correct right now.

File: asteroid/masknn/attention.py [hunk 2]
```diff
@@ -64,8 +58,7 @@ def forward(self, x):
         # mha is seq_len, batch, channels
         # self-attention is applied
         out = self.mha(tomha, tomha, tomha)[0]
-        out = self.pos_wise_ff(out.permute(1, 0, 2)).transpose(1, -1)
-        x = self.dropout(out) + x
+        x = self.dropout(out.permute(1, 2, 0)) + x
```
Comment (SungFeng-Huang) [inline asteroid/masknn/attention.py]: I'm curious whether the dropouts at line 61, 65, 66 shared the same self.dropout would cause any problem?

Comment (mpariente) [reply asteroid/masknn/attention.py]: It's a wrapper above a function and has no state, so it doesn't case any problem, I asked myself the same question in the past :wink: 
```python
    def forward(self, input: Tensor) -> Tensor:
        return F.dropout(input, self.p, self.training, self.inplace)
```

**Comment by popcornell** [comment]
Thank you very much guys, I thought that maybe one can use only two permute operations. But our normalizations actually expects batch, channel, seq so the current version is fine. Merging.
